### DISCUSSION TITLE
What does the database do exactly?

### DISCUSSION THREAD

**Discussion by simplenotezy** [nextauthjs/next-auth#2182] [Help]
## Question 💬

I am trying to integrate Next-auth into an existing database. I have a table with users with whom I wish to log in. They have an e-mail and password.

I find it pretty difficult getting started on this library. Like for instance, database is mentioned several places, but you do not say anything about what it does _exactly_. I imagine it would maybe use it to look up the user e-mail to check if a user exists? Perhaps not? If so, how do I specify what table and field it should look in? I've look in both FAQ and the section related to databases.

Also it's quite confusing that a basic email & password provider is not more documented. I am aware that it might not be best practise (so is magic-password-less logins either I've read), but it is however, part of my scope, and must be a pretty standard requiremenet, at least still in 2021? ;)

Hope to receive some clarification here.

**Comment by simplenotezy** [comment]
Just tested out some more. It's all very confusing. Now I've added my database URL (mongodb). I can login successfully, but upon inspecting the database, nothing has changed. I really think some more documentation is needed as to what the database integration does exactly, because it's pretty confusing.

**Comment by balazsorban44** [comment]
It is our belief that people shouldn't implement their own auth solutions, so we have a limited built-in support, but it is fully possible to integrate with your backend, check these for more info:


https://next-auth.js.org/faq#does-nextauthjs-support-signing-in-with-a-username-and-password

https://next-auth.js.org/providers/credentials

NextAuth.js supports to types of persisting mechanisms. JWT sessions and databases. In vas eof JWT, user data is persisted in a signed, optionally encrypted JWT token in a cookie.

Some features (like email login) are only possible with a database though. (we have to persist the login token somewhere)

in case of a database, you can still decide to use a JWT cookie session. 

The DB tables are for User, Account (think of OAuth profiles connected together through a user table), Session and VerificationRequest

Here is a list of adapters we currently try to support

https://next-auth.js.org/adapters/overview

As an example, here is a schema for a Prisma adapter, it should hopefully give you an idea what is required:
https://next-auth.js.org/adapters/prisma#setup

If you still struggle, please tell us which part of the docs is insufficient, or better yet, please open a PR with your suggestions! Thank you

**Reply by simplenotezy** [reply]
Hey @balazsorban44,

Thanks for your reply, and for the provided links.

Even though I have enabled `jwt:true` and also set a database URL, I am not sure what to do from here, in order to support the database method. We use mongodb for our setup. What other configuration do I need?

I'd like to specify a specific collection/table where we keep our user records.

I am curious, what happens if user signs up/logs in with say GitHub/Google. Will a user record then also be created, if we have enabled the database method?
